### PR TITLE
Use public interface to test `logical_env()`

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
cf. https://github.com/r-lib/lintr/issues/1692

### CODE REVIEW

File: tests/testthat/test-ci.R [hunk 1]
```diff
@@ -1,5 +1,4 @@
 test_that("GitHub Actions functionality works", {
-  # imitate being on GHA whether or not we are
```
Comment (IndrajeetPatil) [inline tests/testthat/test-ci.R:2]: Comment repeats code.

File: tests/testthat/test-ci.R [hunk 2]
```diff
@@ -26,8 +24,21 @@ test_that("GitHub Actions functionality works in a subdirectory", {
...
+  {
+    withr::local_envvar(list(GITHUB_ACTIONS = "true", LINTR_ERROR_ON_LINT = env_var_value))
+    withr::local_options(lintr.rstudio_source_markers = FALSE)
+    tmp <- withr::local_tempfile(lines = "x <- 1:nrow(y)")
+
+    l <- lint(tmp)
+
+    expect_output(print(l), "::warning file", fixed = TRUE)
+  },
+  env_var_value = c("", "F", NA),
```
Comment (IndrajeetPatil) [inline tests/testthat/test-ci.R:38]: We were also testing `NULL`; I am not sure how to incorporate that in this parametrized test suite.

Comment (AshesITR) [reply tests/testthat/test-ci.R]: We are no longer testing `"true"` in this branch, are we?

Comment (AshesITR) [reply tests/testthat/test-ci.R]: Is NULL the equivalent of Sys.unsetenv()?
If so, there is an observable difference in behavior to `""`, so I wouldn't remove the test case.

Comment (IndrajeetPatil) [reply tests/testthat/test-ci.R]: Sounds good. Added a test case for `NULL` as well.

File: tests/testthat/test-ci.R [hunk 3]
```diff
@@ -26,14 +24,31 @@ test_that("GitHub Actions functionality works in a subdirectory", {
...
-  l <- lint(tmp)
+    l <- lint(tmp)
 
-  local_mocked_bindings(quit = function(...) cat("Tried to quit.\n"))
-  expect_output(print(l), "::warning file", fixed = TRUE)
-})
+    local_mocked_bindings(quit = function(...) cat("Tried to quit.\n"))
+    expect_output(print(l), "::warning file", fixed = TRUE)
+  },
+  env_var_value = list("T", "true")
```
Comment (IndrajeetPatil) [inline tests/testthat/test-ci.R:39]: Also checking the variants of true for a good measure.
